purchases/src/test/java/com/revenuecat/purchases/PurchasesCoroutinesTest.kt

```diff
@@ -94,4 +95,94 @@ class PurchasesCoroutinesTest : BasePurchasesTest() {
     }
 
     // endregion
+
+    // region awaitOfferings
+    @Test
+    fun `retrieve offerings - Success`() = runTest {
+        mockOfferingsManagerGetOfferings()
+
+        val result = purchases.awaitOfferings()
+
+        verify(exactly = 1) {
+            mockOfferingsManager.getOfferings(
+                appUserId,
+                any(),
+                any(),
+                any(),
+            )
+        }
+        assertThat(result).isNotNull
+    }
+
+    @Test
+    fun `retrieve offerings - Success - offerings match expectations`() = runTest {
+        val offerings = mockOfferingsManagerGetOfferings()
+
+        val result = purchases.awaitOfferings()
+
+        verify(exactly = 1) {
+            mockOfferingsManager.getOfferings(
+                appUserId,
+                any(),
+                any(),
+                any(),
+            )
+        }
+        assertThat(result).isNotNull
+        assertThat(result).isEqualTo(offerings)
+    }
+
+    @Test
+    fun `retrieve offerings - Error`() = runTest {
+        val error = PurchasesError(PurchasesErrorCode.UnknownError, "Offerings error")
+        mockOfferingsManagerGetOfferings(error)
+
+        var result: Offerings? = null
+        var exception: Throwable? = null
+        runCatching {
+            result = purchases.awaitOfferings()
+        }.onFailure {
+            exception = it
+        }
+
+        verify(exactly = 1) {
+            mockOfferingsManager.getOfferings(
+                appUserId,
+                any(),
+                any(),
+                any(),
+            )
+        }
+        assertThat(result).isNull()
+        assertThat(exception).isNotNull
```

Comment:
Is it possible to check the exact exception?

Reply:
the next test checks for that, I can consolidate them into one test? Or you think it's better to keep them separate

Reply:
I’d combine them if that’s the only difference.